desktop/autoUpdateListeners.js

```diff
@@ -0,0 +1,24 @@
+import { ipcMain } from 'electron';
+import { autoUpdater } from 'electron-updater';
+import type { UpdateCheckResult } from 'electron-updater';
+import { ipcConsts } from '../app/vars';
+
+const subscribeToAutoUpdateListeners = ({ mainWindow }) => {
+  autoUpdater.on('error', (error) => mainWindow.webContents.send(ipcConsts.WALLET_UPDATE_ERROR, { error }));
+
+  autoUpdater.once('update-downloaded', () => mainWindow.webContents.send(ipcConsts.DOWNLOAD_UPDATE_COMPLETED));
+
+  ipcMain.on(ipcConsts.QUIT_AND_UPDATE, () => autoUpdater.quitAndInstall());
+
+  ipcMain.on(ipcConsts.CHECK_WALLET_UPDATE, async () => {
+    try {
+      const updateCheckResult: UpdateCheckResult = await autoUpdater.checkForUpdates();
```

Comment:
documentation doesn't state that this func call can throw error, look like errors are thrown through error event
and there is an error listener already